### DISCUSSION TITLE
How to disable /gateway path ?

### DISCUSSION THREAD

**Discussion by sucheep-s** [Kong/kong#7761] [Help]
Hi, I'm using a Kong with 2.3.3 version. After I configured everything. I've found that the path **/gateway** is being exposed to public. It displays a lot of configuration to public. I don't know where to disable it. I try to find in inside **kong.conf** but not found. Please help. Thanks.

**Comment by bungle** [comment]
@sucheep-s, I am not sure what you are talking about. Afaik we don't have such a path. Can you show us the output?

**Reply by sucheep-s** [reply]
When I enter to mydomain.com/gateway It will show configs like below. I do not know how it happens. It's found by a tester. 
ps. I'm very new to Kong. 

`{
plugins: {
enabled_in_cluster: [ ],
available_on_server: {
grpc-web: true,
correlation-id: true,
pre-function: true,
cors: true,
rate-limiting: true,
loggly: true,
hmac-auth: true,
zipkin: true,
request-size-limiting: true,
azure-functions: true,
request-transformer: true,
oauth2: true,
response-transformer: true,
ip-restriction: true,
statsd: true,
jwt: true,
proxy-cache: true,
basic-auth: true,
key-auth: true,
http-log: true,
session: true,
datadog: true,
tcp-log: true,
prometheus: true,
post-function: true,
ldap-auth: true,
acl: true,
grpc-gateway: true,
file-log: true,
syslog: true,
udp-log: true,
response-ratelimiting: true,
aws-lambda: true,
bot-detection: true,
acme: true,
request-termination: true
}
},
tagline: "Welcome to kong",
configuration: {
pluginserver_names: { },
cassandra_read_consistency: "ONE",
admin_listen: [
"127.0.0.1:8001 reuseport backlog=16384",
"127.0.0.1:8444 http2 ssl reuseport backlog=16384"
],
proxy_access_log: "logs/access.log",
upstream_keepalive_max_requests: 100,
nginx_conf: "/usr/local/kong/nginx.conf",
cassandra_username: "kong",
nginx_admin_client_body_buffer_size: "10m",
admin_ssl_cert_key: [
"/usr/local/kong/ssl/admin-kong-default.key",
"/usr/local/kong/ssl/admin-kong-default-ecdsa.key"
],
status_ssl_cert: { },
untrusted_lua: "sandbox",
nginx_http_upstream_directives: { },
nginx_main_daemon: "on",
nginx_http_ssl_dhparam: "ffdhe2048",
stream_proxy_ssl_enabled: false,
nginx_user: "kong kong",
pg_semaphore_timeout: 60000,
proxy_listen: [
"0.0.0.0:80 reuseport backlog=16384",
"0.0.0.0:443 http2 ssl reuseport backlog=16384"
],
nginx_main_user: "kong kong",
client_ssl_cert_default: "/usr/local/kong/ssl/kong-default.crt",
untrusted_lua_sandbox_requires: { },
dns_no_sync: false,
db_update_propagation: 0,
nginx_stream_ssl_session_tickets: "on",
nginx_admin_client_max_body_size: "10m",
ssl_prefer_server_ciphers: "on",
status_ssl_cert_key_default_ecdsa: "/usr/local/kong/ssl/status-kong-default-ecdsa.key",
headers: [
"server_tokens",
"latency_tokens"
],
nginx_http_client_max_body_size: "0",
status_ssl_enabled: false,
status_listen: [
"off"
],
nginx_status_directives: { },
cluster_control_plane: "127.0.0.1:8005",
nginx_stream_ssl_dhparam: "ffdhe2048",
nginx_http_ssl_prefer_server_ciphers: "off",
pg_database: "kong",
nginx_http_client_body_buffer_size: "8k",
admin_acc_logs: "/usr/local/kong/logs/admin_access.log",
cassandra_refresh_frequency: 60,
nginx_pid: "/usr/local/kong/pids/nginx.pid",
nginx_main_worker_rlimit_nofile: "auto",
cassandra_contact_points: [
"127.0.0.1"
],
ssl_cert_key_default_ecdsa: "/usr/local/kong/ssl/kong-default-ecdsa.key",
db_cache_warmup_entities: [
"services"
],
enabled_headers: {
latency_tokens: true,
X-Kong-Response-Latency: true,
Server: true,
X-Kong-Admin-Latency: true,
X-Kong-Upstream-Status: false,
Via: true,
X-Kong-Proxy-Latency: true,
server_tokens: true,
X-Kong-Upstream-Latency: true
},
nginx_http_ssl_protocols: "TLSv1.2 TLSv1.3",
upstream_keepalive_idle_timeout: 60,
db_cache_ttl: 0,
nginx_events_multi_accept: "on",
admin_ssl_cert_default: "/usr/local/kong/ssl/admin-kong-default.crt",
pg_ssl: false,
status_access_log: "off",
cluster_listeners: [
{
listener: "0.0.0.0:8005",
proxy_protocol: false,
reuseport: false,
backlog=%d+: false,
deferred: false,
ssl: false,
ip: "0.0.0.0",
port: 8005,
http2: false,
bind: false
}
],
ssl_protocols: "TLSv1.1 TLSv1.2 TLSv1.3",
kong_env: "/usr/local/kong/.kong_env",
cassandra_schema_consensus_timeout: 60000,
ssl_dhparam: "ffdhe2048",
untrusted_lua_sandbox_environment: { },
admin_ssl_cert_key_default: "/usr/local/kong/ssl/admin-kong-default.key",
ssl_session_timeout: "1d",
admin_ssl_cert_key_default_ecdsa: "/usr/local/kong/ssl/admin-kong-default-ecdsa.key",
cassandra_repl_factor: 1,
nginx_stream_ssl_protocols: "TLSv1.2 TLSv1.3",
port_maps: { },
pg_port: 5432,
cassandra_keyspace: "kong",
ssl_cert_default: "/usr/local/kong/ssl/kong-default.crt",
nginx_http_ssl_session_timeout: "1d",
nginx_upstream_directives: { },
upstream_keepalive_pool_size: 60,
worker_consistency: "strict",
nginx_stream_ssl_session_timeout: "1d",
cluster_data_plane_purge_delay: 1209600,
admin_ssl_enabled: true,
trusted_ips: { },
loaded_plugins: {
grpc-web: true,
correlation-id: true,
pre-function: true,
cors: true,
rate-limiting: true,
loggly: true,
hmac-auth: true,
zipkin: true,
request-size-limiting: true,
azure-functions: true,
request-transformer: true,
oauth2: true,
prometheus: true,
syslog: true,
statsd: true,
jwt: true,
proxy-cache: true,
basic-auth: true,
key-auth: true,
http-log: true,
session: true,
datadog: true,
tcp-log: true,
acme: true,
post-function: true,
bot-detection: true,
acl: true,
grpc-gateway: true,
response-transformer: true,
ip-restriction: true,
udp-log: true,
response-ratelimiting: true,
aws-lambda: true,
file-log: true,
ldap-auth: true,
request-termination: true
},
nginx_supstream_directives: { },
ssl_cert_key: [
"/etc/ssl/private.key"
],
host_ports: { },
pg_user: "kong",
mem_cache_size: "128m",
cassandra_data_centers: [
"dc1:2",
"dc2:3"
],
nginx_admin_directives: [
{
value: "10m",
name: "client_body_buffer_size"
},
{
value: "10m",
name: "client_max_body_size"
}
],
nginx_http_directives: [
{
value: "8k",
name: "client_body_buffer_size"
},
{
value: "0",
name: "client_max_body_size"
},
{
value: "prometheus_metrics 5m",
name: "lua_shared_dict"
},
{
value: "/usr/local/kong/ssl/ffdhe2048.pem",
name: "ssl_dhparam"
},
{
value: "off",
name: "ssl_prefer_server_ciphers"
},
{
value: "TLSv1.2 TLSv1.3",
name: "ssl_protocols"
},
{
value: "on",
name: "ssl_session_tickets"
},
{
value: "1d",
name: "ssl_session_timeout"
}
],
pg_host: "127.0.0.1",
nginx_kong_stream_conf: "/usr/local/kong/nginx-kong-stream.conf",
nginx_proxy_real_ip_header: "X-Real-IP",
go_plugins_dir: "off",
cluster_listen: [
"0.0.0.0:8005"
],
cassandra_write_consistency: "ONE",
dns_order: [
"LAST",
"SRV",
"A",
"CNAME"
],
dns_error_ttl: 1,
cluster_mtls: "shared",
lua_ssl_verify_depth: 1,
ssl_cert_default_ecdsa: "/usr/local/kong/ssl/kong-default-ecdsa.crt",
cassandra_port: 9042,
dns_stale_ttl: 4,
pg_timeout: 60000,
ssl_ciphers: "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384",
pg_ro_ssl_verify: false,
database: "postgres",
cassandra_lb_policy: "RequestRoundRobin",
proxy_listeners: [
{
listener: "0.0.0.0:80 reuseport backlog=16384",
proxy_protocol: false,
reuseport: true,
deferred: false,
ssl: false,
ip: "0.0.0.0",
backlog=16384: true,
http2: false,
port: 80,
bind: false
},
{
listener: "0.0.0.0:443 ssl http2 reuseport backlog=16384",
proxy_protocol: false,
reuseport: true,
deferred: false,
ssl: true,
ip: "0.0.0.0",
backlog=16384: true,
http2: true,
port: 443,
bind: false
}
],
pg_password: "******",
admin_ssl_cert_default_ecdsa: "/usr/local/kong/ssl/admin-kong-default-ecdsa.crt",
nginx_worker_processes: "auto",
stream_listen: [
"off"
],
nginx_http_status_directives: { },
plugins: [
"bundled"
],
nginx_acc_logs: "/usr/local/kong/logs/access.log",
nginx_daemon: "on",
nginx_proxy_directives: [
{
value: "X-Real-IP",
name: "real_ip_header"
},
{
value: "off",
name: "real_ip_recursive"
}
],
lua_package_path: "./?.lua;./?/init.lua;",
go_pluginserver_exe: "/usr/local/bin/go-pluginserver",
dns_resolver: { },
db_update_frequency: 5,
worker_state_update_frequency: 5,
nginx_main_worker_processes: "auto",
cassandra_ssl_verify: false,
pg_max_concurrent_queries: 0,
real_ip_recursive: "off",
nginx_events_worker_connections: "auto",
client_ssl: false,
proxy_ssl_enabled: true,
role: "traditional",
error_default_type: "text/plain",
lua_socket_pool_size: 30,
dns_hostsfile: "/etc/hosts",
prefix: "/usr/local/kong",
db_resurrect_ttl: 30,
admin_access_log: "logs/admin_access.log",
pg_ssl_verify: false,
client_max_body_size: "0",
admin_error_log: "logs/error.log",
nginx_main_directives: [
{
value: "on",
name: "daemon"
},
{
value: "kong kong",
name: "user"
},
{
value: "auto",
name: "worker_processes"
},
{
value: "auto",
name: "worker_rlimit_nofile"
}
],
dns_not_found_ttl: 30,
nginx_http_ssl_session_tickets: "on",
ssl_cipher_suite: "intermediate",
cassandra_timeout: 60000,
cassandra_repl_strategy: "SimpleStrategy",
status_error_log: "logs/status_error.log",
proxy_error_log: "logs/error.log",
kic: false,
nginx_optimizations: true,
nginx_kong_conf: "/usr/local/kong/nginx-kong.conf",
real_ip_header: "X-Real-IP",
status_ssl_cert_key_default: "/usr/local/kong/ssl/status-kong-default.key",
admin_listeners: [
{
listener: "127.0.0.1:8001 reuseport backlog=16384",
proxy_protocol: false,
reuseport: true,
deferred: false,
ssl: false,
ip: "127.0.0.1",
backlog=16384: true,
http2: false,
port: 8001,
bind: false
},
{
listener: "127.0.0.1:8444 ssl http2 reuseport backlog=16384",
proxy_protocol: false,
reuseport: true,
deferred: false,
ssl: true,
ip: "127.0.0.1",
backlog=16384: true,
http2: true,
port: 8444,
bind: false
}
],
status_ssl_cert_default_ecdsa: "/usr/local/kong/ssl/status-kong-default-ecdsa.crt",
ssl_cert: [
"/etc/ssl/ssl_certificate.crt"
],
nginx_proxy_real_ip_recursive: "off",
pg_ro_ssl: false,
nginx_stream_ssl_prefer_server_ciphers: "off",
cassandra_ssl: false,
status_ssl_cert_key: { },
log_level: "notice",
ssl_cert_csr_default: "/usr/local/kong/ssl/kong-default.csr",
client_ssl_cert_key_default: "/usr/local/kong/ssl/kong-default.key",
admin_ssl_cert: [
"/usr/local/kong/ssl/admin-kong-default.crt",
"/usr/local/kong/ssl/admin-kong-default-ecdsa.crt"
],
anonymous_reports: true,
status_listeners: { },
lua_package_cpath: "",
status_ssl_cert_default: "/usr/local/kong/ssl/status-kong-default.crt",
lua_ssl_trusted_certificate: { },
client_body_buffer_size: "8k",
nginx_err_logs: "/usr/local/kong/logs/error.log",
ssl_cert_key_default: "/usr/local/kong/ssl/kong-default.key",
stream_listeners: { },
ssl_session_tickets: "on",
nginx_sproxy_directives: { },
nginx_stream_directives: [
{
value: "stream_prometheus_metrics 5m",
name: "lua_shared_dict"
},
{
value: "/usr/local/kong/ssl/ffdhe2048.pem",
name: "ssl_dhparam"
},
{
value: "off",
name: "ssl_prefer_server_ciphers"
},
{
value: "TLSv1.2 TLSv1.3",
name: "ssl_protocols"
},
{
value: "on",
name: "ssl_session_tickets"
},
{
value: "1d",
name: "ssl_session_timeout"
}
],
nginx_events_directives: [
{
value: "on",
name: "multi_accept"
},
{
value: "auto",
name: "worker_connections"
}
]
},
version: "2.3.3",
pids: {
workers: [
31546,
31547
],
master: 31545
},
node_id: "2d661d83-cbc7-480c-9d5d-842494731da4",
lua_version: "LuaJIT 2.1.0-beta3",
timers: {
pending: 9,
running: 0
},
hostname: "ip-172-31-12-49"
} `

**Comment by sucheep-s** [comment]
More clues, When I check from **curl localhost:8001/routes** There is no /gateway is set up.

**Comment by sucheep-s** [comment]
Something more weird. I access **mydomain.com/gateway/routes**, There's is the service below is set up,

`{
strip_path: true,
tags: null,
updated_at: 1615316963,
destinations: null,
headers: null,
protocols: [
"https"
],
methods: null,
service: {
id: "740fbb3a-6f99-4efd-a3e5-34b7e28d7446"
},
snis: null,
hosts: null,
name: "kong_gateway",
path_handling: "v0",
paths: [
"/gateway"
],
preserve_host: false,
regex_priority: 0,
response_buffering: true,
sources: null,
id: "0129b605-fcef-4c93-90f1-55352effcfa4",
https_redirect_status_code: 302,
request_buffering: true,
created_at: 1615316963
}`

But this will not be shown for localhost:8001
